service/internal/proctelemetry/process_telemetry.go

```diff
@@ -31,100 +48,224 @@ type processMetrics struct {
+		instrument.WithDescription("Cumulative bytes allocated for heap objects (see 'go doc runtime.MemStats.TotalAlloc')"),
+		instrument.WithUnit(unit.Bytes))
+	errors = multierr.Append(errors, err)
+
+	pm.otelSysMem, err = pm.meter.AsyncInt64().Counter(
+		"process_runtime_total_sys_memory_bytes",
+		instrument.WithDescription("Total bytes of memory obtained from the OS (see 'go doc runtime.MemStats.Sys')"),
+		instrument.WithUnit(unit.Bytes))
+	errors = multierr.Append(errors, err)
+
+	pm.otelCpuSeconds, err = pm.meter.AsyncFloat64().Counter(
+		"process_cpu_seconds",
+		instrument.WithDescription("Total CPU user and system time in seconds"),
+		instrument.WithUnit(unit.Milliseconds))
+	errors = multierr.Append(errors, err)
+
+	pm.otelRssMemory, err = pm.meter.AsyncInt64().Counter(
+		"process_memory_rss",
+		instrument.WithDescription("Total physical memory (resident set size)"),
+		instrument.WithUnit(unit.Bytes))
+	errors = multierr.Append(errors, err)
+
+	return errors
+}
+
+func (pm *processMetrics) recordWithOtel(ctx context.Context) error {
+	var errors, err error
+	err = pm.meter.RegisterCallback([]instrument.Asynchronous{pm.otelProcessUptime}, func(ctx context.Context) {
+		processTimeMs := 1000 * pm.updateProcessUptime()
+		pm.otelProcessUptime.Observe(ctx, processTimeMs)
+	})
+	errors = multierr.Append(errors, err)
+
+	err = pm.meter.RegisterCallback([]instrument.Asynchronous{pm.otelAllocMem}, func(ctx context.Context) {
+		pm.otelAllocMem.Observe(ctx, pm.updateAllocMem())
+	})
+	errors = multierr.Append(errors, err)
+
+	err = pm.meter.RegisterCallback([]instrument.Asynchronous{pm.otelTotalAllocMem}, func(ctx context.Context) {
+		pm.otelTotalAllocMem.Observe(ctx, pm.updateTotalAllocMem())
+	})
+	errors = multierr.Append(errors, err)
+
+	err = pm.meter.RegisterCallback([]instrument.Asynchronous{pm.otelSysMem}, func(ctx context.Context) {
+		pm.otelSysMem.Observe(ctx, pm.updateSysMem())
+	})
+	errors = multierr.Append(errors, err)
+
+	err = pm.meter.RegisterCallback([]instrument.Asynchronous{pm.otelCpuSeconds}, func(ctx context.Context) {
+		cpuMs := 1000 * pm.updateCPUSeconds()
+		pm.otelCpuSeconds.Observe(ctx, cpuMs)
+	})
+	errors = multierr.Append(errors, err)
+
+	err = pm.meter.RegisterCallback([]instrument.Asynchronous{pm.otelRssMemory}, func(ctx context.Context) {
+		pm.otelRssMemory.Observe(ctx, pm.updateRSSMemory())
+	})
+	errors = multierr.Append(errors, err)
+
+	return errors
+}
+
+func (pm *processMetrics) createOCMetrics(registry *ocmetric.Registry) error {
+	var errors, err error
+
 	pm.processUptime, err = registry.AddFloat64DerivedCumulative(
 		"process/uptime",
-		metric.WithDescription("Uptime of the process"),
-		metric.WithUnit(stats.UnitSeconds))
-	if err != nil {
-		return err
-	}
-	if err = pm.processUptime.UpsertEntry(pm.updateProcessUptime); err != nil {
-		return err
-	}
+		ocmetric.WithDescription("Uptime of the process"),
+		ocmetric.WithUnit(stats.UnitSeconds))
+	errors = multierr.Append(errors, err)
```

Comment:
Although I like these changes, I would suggest you to not do them in this PR. Keep the changes to the current code as little as possible.

I would even suggest revert the change from `metric -> ocmetric` and instead rename the  otel `metric` to `otelmetric`

The less changes, the less conflict you'll have during reviews. Also think that this code will once be deleted after the migration!

Reply:
That makes sense to me, I will revert these changes.